src/Register.ts

```diff
@@ -51,27 +39,28 @@ export class Register {
    *
    * Therefore, we have to make use of static props vs instance props and use getters and setters to merge these together.
    */
-  static hooks = getBuiltinHooks();
+  static hooks: Required<Hooks> = getBuiltinHooks();
 
   /** @type {Required<Hooks>} */
-  get hooks() {
-    const ctor = /** @type {typeof Register} */ (this.constructor);
-    return deepmerge(ctor.hooks, this._hooks ?? {});
+  get hooks(): Required<Hooks> {
+    return deepmerge((this.constructor as any).hooks, this._hooks ?? {});
```

Comment:
https://github.com/microsoft/TypeScript/issues/3841

Reply:
Yeah this issue is a nuisance but I don't see how typecasting to `any` is better than typecasting to `typeof Register` which is actually the proper cast and what it should be implicitly typed as if that TS issue were solved.